codespell_lib/data/dictionary.txt

```diff
@@ -14558,8 +14558,8 @@ hertzs->hertz
 hesiate->hesitate
 hesistant->hesitant
 hestiate->hesitate
-heterogenous->heterogeneous
-hetrogenous->heterogeneous
+hetrogeneous->heterogeneous
```

Comment:
Maybe this should suggest the other one too, with *no* second?

Reply:
I'm not sure that's warranted. My impression is that "heterogeneous" is significantly more common than "heterogeneous", and the data seems to corroborate it ([google ngram viewer](https://books.google.com/ngrams/graph?content=heterogenous%2Cheterogeneous); [4M](https://www.google.com/search?q=heterogenous) vs [111M](https://www.google.com/search?q=heterogeneous) search results). I'd even venture that "heterogenous" might belong in the rare dictionary.

Reply:
I'm just always a bit wary about it automatically fixing stuff; if you've made a typo and there is another plausible word, you could just has easily have made more of/a different typo!

Reply:
> I'd even venture that "heterogenous" might belong in the rare dictionary.

The rare dictionary is for real words that may be typos of other words, e.g. bellow->below, so we could add this to rare:
`heterogenous->heterogeneous`

I think it allows three way corrections (i.e. having typos for them individually at the same time).

Reply:
Agreed. So just to make sure we're aligned, what you're suggesting is:

- add `heterogenous->heterogeneous` to the rare dictionary
- _not_ change `hetrogeneous->heterogeneous` into `hetrogeneous->heterogeneous,heterogenous,` in this line

Is that right?

Reply:
> Agreed. So just to make sure we're aligned, what you're suggesting is:
> 
> * add `heterogenous->heterogeneous` to the rare dictionary

Yes, assuming it doesn't get confused by the resultant chain of hetrogeneous->heterogenous->heterogeneous

> * _not_ change `hetrogeneous->heterogeneous` into `hetrogeneous->heterogeneous,heterogenous,` in this line

I think (well I think that I think), that we should suggest both and not automatically convert hetrogeneous to heterogeneous. Because hetrogenEous to hetErogenous is only the move of one E a few places.

Reply:
> Because hetrogenEous to hetErogenous is only the move of one E a few places.

Hmm, not sure I agree with that reasoning. That's a very mechanistic way to look at the issue — other than accidentally swapping the order of a pair of letters, it's not common for a human to move a letter several places across a word by accident. Semantically-wise, I would say those correspond to different, independent typos: "hetro-" instead of "hetero-", and "-genous" instead of "-geneous". 

With that in mind, IMO it's quite likely that one would (1) commit both typos simultaneously, and (2) actually mean the much rarer "-genous" suffix as opposed to "-geneous". I'd say catering to that case is being excessively cautious and making the tool more cumbersome to the vast majority of users.

Reply:
Okay, fair enough. I'm not hugely heavily wedded to it for something this niche.

Reply:
Ok, I've updated the PR.